source/routing/raptor_api.cpp

```diff
@@ -1144,5 +1145,63 @@ pbnavitia::Response make_isochrone(RAPTOR &raptor,
     return pb_creator.get_response();
 }
 
+void add_graphical_isochron(const type::MultiPolygon& shape, PbCreator& pb_creator) {
+    auto pb_isochron = pb_creator.add_graphical_isochrons();
+    auto pb_polys = pb_isochron->mutable_geojson();
+    for (const auto& polygon: shape) {
+        auto p = pb_polys->add_polygons();
+        auto lo = p->mutable_outer();
+        for (const auto coord: polygon.outer()) {
+            auto c = lo->add_coordinates();
+            c->set_lon(coord.lon());
+            c->set_lat(coord.lat());
+        }
+        for (const auto inner: polygon.inners()) {
+            auto li = p->add_inners();
+            for (const auto coord: inner) {
+                auto c = li->add_coordinates();
+                c->set_lon(coord.lon());
+                c->set_lat(coord.lat());
+            }
+        }
+    }
+}
+
+pbnavitia::Response make_graphical_isochrone(RAPTOR &raptor,
+                                             const boost::posix_time::ptime& current_datetime,
+                                             type::EntryPoint origin,
+                                             const uint64_t departure_datetime,
+                                             int max_duration,
+                                             uint32_t max_transfers,
+                                             const type::AccessibiliteParams& accessibilite_params,
+                                             const std::vector<std::string>& forbidden,
+                                             bool clockwise,
+                                             const nt::RTLevel rt_level,
+                                             georef::StreetNetwork & worker) {
+
+    PbCreator pb_creator(raptor.data, current_datetime, null_time_period);
+    bt::ptime datetime;
+    auto tmp_datetime = parse_datetimes(raptor, {departure_datetime}, pb_creator, clockwise);
+    if(pb_creator.has_error() || tmp_datetime.size() == 0 ||
+            pb_creator.has_response_type(pbnavitia::DATE_OUT_OF_BOUNDS)) {
+        return pb_creator.get_response();
+    }
+    datetime = tmp_datetime.front();
+    worker.init(origin);
+    auto departures = get_stop_points(origin, raptor.data, worker);
+
+    if(departures.size() == 0){
```

Comment:
It's better to use `! empty()` because it's more what you want to check and on some containers the `.size()` computation can be expensive

and you should put spaces between the around `(` and `)`